### PR TITLE
'istioctl get all' implementation

### PR DESCRIPTION
**Pull request by esnible**
Resolves https://github.com/istio/istio/issues/5502

### CODE REVIEW

File: istioctl/cmd/istioctl/main.go [hunk 1]
```diff
@@ -298,8 +311,8 @@ istioctl get virtualservice bookinfo
 					strings.Join(supportedTypes(configClient), ", "))
 			}
 
-			typ, err := protoSchema(configClient, strings.ToLower(args[0]))
```
Comment (ijsnellf) [inline istioctl/cmd/istioctl/main.go]: `strings.ToLower(args[0])` can be safely removed here?

Comment (ayj) [reply istioctl/cmd/istioctl/main.go]: istioctl and kubectl support case insensitive type names for get. ToLower is needed here or we should add it to `protoSchema`

File: istioctl/cmd/istioctl/main.go [hunk 2]
```diff
@@ -298,8 +311,8 @@ istioctl get virtualservice bookinfo
...
-			if err != nil {
+			typs, err := protoSchema(configClient, args[0], len(args) == 1)
```
Comment (ayj) [inline istioctl/cmd/istioctl/main.go]: `protoSchema` shouldn't be overloaded to handle input/flag validation here. Lines 320-323 should catch the invalid allowAll/all combination.

Comment (ayj) [reply istioctl/cmd/istioctl/main.go]: `kubectl get all --all-namespaces` works. Not sure why we're disallowing this combination for istioctl.

**Pull request merged**